### PR TITLE
feat: [AXIMST-15] Course unit - add new component section

### PR DESCRIPTION
**Pull request by ihor-romaniuk**
**Description:** add new component section

**Youtrack:** https://youtrack.raccoongang.com/issue/AXIMST-15

### CODE REVIEW

**Review by ruzniaievdm — APPROVED** [review]

File: src/course-unit/add-component/ComponentIcon.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import PropTypes from 'prop-types';
+import { Icon } from '@edx/paragon';
+import { EditNote as EditNoteIcon } from '@edx/paragon/icons';
+
+import { componentTypeToIconMapping, COMPONENT_ICON_TYPES } from '../constants';
+
+const ComponentIcon = ({ type }) => {
+  const icon = componentTypeToIconMapping[type] || EditNoteIcon;
```
Comment (PKulkoRaccoonGang) [inline src/course-unit/add-component/ComponentIcon.jsx:8]: [fyi] We added new icons in the [pull request for the fix](https://github.com/raccoongang/edx-platform/pull/2490/files#diff-cc20776c84f78638c79199d2b4d0c30c2e296b8dc82dc9b41de8a0706e0905edR183). Could you make sure that the lights will be displayed correctly for you?

File: src/course-unit/add-component/messages.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { defineMessages } from '@edx/frontend-platform/i18n';
+
+const messages = defineMessages({
+  title: {
+    id: 'course-authoring.course-unit.add.component.title',
```
Comment (PKulkoRaccoonGang) [inline src/course-unit/add-component/messages.js:5]: [important] You added translations, but I don't see any changes in the key files (`i18n` directory). 
Please add 🥇

**Review by monteri — APPROVED** [review]
LGTM waiting for fixing comments

File: src/course-unit/add-component/AddComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { useSelector } from 'react-redux';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Button } from '@edx/paragon';
+
+import { getCourseSectionVertical } from '../data/selectors';
+import ComponentIcon from './ComponentIcon';
+
```
[the remark this answers is not in the document]
Comment (ihor-romaniuk) [reply src/course-unit/add-component/AddComponent.jsx:7]: Removed

File: src/course-unit/add-component/messages.js (hunk 2 above)
Comment (ihor-romaniuk) [reply src/course-unit/add-component/messages.js:5]: Added

**Review by PKulkoRaccoonGang — APPROVED** [review]

**Pull request merged**